### PR TITLE
feat(repo): Add Vite React Typescript playground app

### PR DESCRIPTION
**Pull request by anagstef**
## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [x] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [x] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

This PR adds a Clerk + Vite + React + TypeScript playground.
Also, it resolves the `ReferenceError: global is not defined` error on Vite + React projects.

### CODE REVIEW

**Comment by anagstef** [comment]
!snapshot

**Comment by anagstef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.6-snapshot.73a3055 |
| @clerk/chrome-extension | 0.3.17-snapshot.73a3055 |
| @clerk/clerk-js | 4.51.0-snapshot.73a3055 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.8-snapshot.73a3055 |
| @clerk/fastify | 0.5.5-snapshot.73a3055 |
| gatsby-plugin-clerk | 4.3.17-snapshot.73a3055 |
| @clerk/localizations | 1.22.0-snapshot.73a3055 |
| @clerk/nextjs | 4.21.11-snapshot.73a3055 |
| @clerk/clerk-react | 4.20.6-snapshot.73a3055 |
| @clerk/remix | 2.6.14-snapshot.73a3055 |
| @clerk/clerk-sdk-node | 4.10.13-snapshot.73a3055 |
| @clerk/shared | 0.20.0-snapshot.73a3055 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.46.1-snapshot.73a3055 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.6-snapshot.73a3055
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.17-snapshot.73a3055
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.51.0-snapshot.73a3055
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.8-snapshot.73a3055
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.5-snapshot.73a3055
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.17-snapshot.73a3055
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.0-snapshot.73a3055
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.11-snapshot.73a3055
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.20.6-snapshot.73a3055
```
```sh
# @clerk/remix
npm i @clerk/remix@2.6.14-snapshot.73a3055
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.13-snapshot.73a3055
```
```sh
# @clerk/shared
npm i @clerk/shared@0.20.0-snapshot.73a3055
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.46.1-snapshot.73a3055
```

**Comment by anagstef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.6-snapshot.bae5a19 |
| @clerk/chrome-extension | 0.3.17-snapshot.bae5a19 |
| @clerk/clerk-js | 4.51.0-snapshot.bae5a19 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.8-snapshot.bae5a19 |
| @clerk/fastify | 0.5.5-snapshot.bae5a19 |
| gatsby-plugin-clerk | 4.3.17-snapshot.bae5a19 |
| @clerk/localizations | 1.22.0-snapshot.bae5a19 |
| @clerk/nextjs | 4.21.11-snapshot.bae5a19 |
| @clerk/clerk-react | 4.20.6-snapshot.bae5a19 |
| @clerk/remix | 2.6.14-snapshot.bae5a19 |
| @clerk/clerk-sdk-node | 4.10.13-snapshot.bae5a19 |
| @clerk/shared | 0.20.0-snapshot.bae5a19 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.46.1-snapshot.bae5a19 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.6-snapshot.bae5a19
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.17-snapshot.bae5a19
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.51.0-snapshot.bae5a19
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.8-snapshot.bae5a19
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.5-snapshot.bae5a19
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.17-snapshot.bae5a19
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.0-snapshot.bae5a19
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.11-snapshot.bae5a19
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.20.6-snapshot.bae5a19
```
```sh
# @clerk/remix
npm i @clerk/remix@2.6.14-snapshot.bae5a19
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.13-snapshot.bae5a19
```
```sh
# @clerk/shared
npm i @clerk/shared@0.20.0-snapshot.bae5a19
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.46.1-snapshot.bae5a19
```

**Comment by anagstef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.6-snapshot.c7c8096 |
| @clerk/chrome-extension | 0.3.17-snapshot.c7c8096 |
| @clerk/clerk-js | 4.51.0-snapshot.c7c8096 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.8-snapshot.c7c8096 |
| @clerk/fastify | 0.5.5-snapshot.c7c8096 |
| gatsby-plugin-clerk | 4.3.17-snapshot.c7c8096 |
| @clerk/localizations | 1.22.0-snapshot.c7c8096 |
| @clerk/nextjs | 4.21.11-snapshot.c7c8096 |
| @clerk/clerk-react | 4.20.6-snapshot.c7c8096 |
| @clerk/remix | 2.6.14-snapshot.c7c8096 |
| @clerk/clerk-sdk-node | 4.10.13-snapshot.c7c8096 |
| @clerk/shared | 0.20.0-snapshot.c7c8096 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.46.1-snapshot.c7c8096 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.6-snapshot.c7c8096
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.17-snapshot.c7c8096
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.51.0-snapshot.c7c8096
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.8-snapshot.c7c8096
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.5-snapshot.c7c8096
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.17-snapshot.c7c8096
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.0-snapshot.c7c8096
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.11-snapshot.c7c8096
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.20.6-snapshot.c7c8096
```
```sh
# @clerk/remix
npm i @clerk/remix@2.6.14-snapshot.c7c8096
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.13-snapshot.c7c8096
```
```sh
# @clerk/shared
npm i @clerk/shared@0.20.0-snapshot.c7c8096
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.46.1-snapshot.c7c8096
```

**Comment by anagstef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.6-snapshot.1819c97 |
| @clerk/chrome-extension | 0.3.17-snapshot.1819c97 |
| @clerk/clerk-js | 4.51.0-snapshot.1819c97 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.8-snapshot.1819c97 |
| @clerk/fastify | 0.5.5-snapshot.1819c97 |
| gatsby-plugin-clerk | 4.3.17-snapshot.1819c97 |
| @clerk/localizations | 1.22.0-snapshot.1819c97 |
| @clerk/nextjs | 4.21.11-snapshot.1819c97 |
| @clerk/clerk-react | 4.20.6-snapshot.1819c97 |
| @clerk/remix | 2.6.14-snapshot.1819c97 |
| @clerk/clerk-sdk-node | 4.10.13-snapshot.1819c97 |
| @clerk/shared | 0.20.0-snapshot.1819c97 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.46.1-snapshot.1819c97 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.6-snapshot.1819c97
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.17-snapshot.1819c97
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.51.0-snapshot.1819c97
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.8-snapshot.1819c97
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.5-snapshot.1819c97
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.17-snapshot.1819c97
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.0-snapshot.1819c97
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.11-snapshot.1819c97
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.20.6-snapshot.1819c97
```
```sh
# @clerk/remix
npm i @clerk/remix@2.6.14-snapshot.1819c97
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.13-snapshot.1819c97
```
```sh
# @clerk/shared
npm i @clerk/shared@0.20.0-snapshot.1819c97
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.46.1-snapshot.1819c97
```

**Comment by clerk-cookie** [comment]
Hey @anagstef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.23.6-snapshot.c9a29e3 |
| @clerk/chrome-extension | 0.3.17-snapshot.c9a29e3 |
| @clerk/clerk-js | 4.51.0-snapshot.c9a29e3 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.8-snapshot.c9a29e3 |
| @clerk/fastify | 0.5.5-snapshot.c9a29e3 |
| gatsby-plugin-clerk | 4.3.17-snapshot.c9a29e3 |
| @clerk/localizations | 1.22.0-snapshot.c9a29e3 |
| @clerk/nextjs | 4.21.11-snapshot.c9a29e3 |
| @clerk/clerk-react | 4.20.6-snapshot.c9a29e3 |
| @clerk/remix | 2.6.14-snapshot.c9a29e3 |
| @clerk/clerk-sdk-node | 4.10.13-snapshot.c9a29e3 |
| @clerk/shared | 0.20.0-snapshot.c9a29e3 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.46.1-snapshot.c9a29e3 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.23.6-snapshot.c9a29e3
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.17-snapshot.c9a29e3
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.51.0-snapshot.c9a29e3
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.8-snapshot.c9a29e3
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.5.5-snapshot.c9a29e3
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.3.17-snapshot.c9a29e3
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.22.0-snapshot.c9a29e3
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.21.11-snapshot.c9a29e3
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.20.6-snapshot.c9a29e3
```
```sh
# @clerk/remix
npm i @clerk/remix@2.6.14-snapshot.c9a29e3
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.10.13-snapshot.c9a29e3
```
```sh
# @clerk/shared
npm i @clerk/shared@0.20.0-snapshot.c9a29e3
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.46.1-snapshot.c9a29e3
```

File: playground/vite-react-ts/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import reactLogo from './assets/react.svg'
+import viteLogo from '/vite.svg'
+import clerkLogo from '/clerk.svg'
+import './App.css'
+import { ClerkProvider, RedirectToSignIn, SignIn, SignUp, SignedIn, SignedOut, UserButton, UserProfile, useUser } from "@clerk/clerk-react";
+import { BrowserRouter, Route, Routes, useNavigate } from "react-router-dom";
+
+if (!import.meta.env.VITE_REACT_APP_CLERK_PUBLISHABLE_KEY) {
+  throw new Error("Missing Publishable Key")
+}
```
Comment (dimkl) [inline playground/vite-react-ts/src/App.tsx:10]: ❓ wouldn't the `ClerkProvider` raise an error if no publishableKey or frontendApi exists? I think we should omit this check.

**Review by dimkl — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]
🎉

File: playground/vite-react-ts/src/App.tsx (hunk 1 above)
Comment (nikosdouvlis) [reply playground/vite-react-ts/src/App.tsx]: We don't need this check

File: playground/vite-react-ts/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  );
+}
+
+function ClerkProviderWithRoutes() {
+  const navigate = useNavigate();
+
+  return (
+    <ClerkProvider
+      publishableKey={clerkPubKey}
+      navigate={(to) => navigate(to)}
```
[the remark this answers is not in the document]
Comment (nikosdouvlis) [reply playground/vite-react-ts/src/App.tsx]: We need access to the router of the parent app so we can perform the internal navigations without full page reloads

File: playground/vite-react-ts/src/App.tsx (hunk 1 above)
Comment (anagstef) [reply playground/vite-react-ts/src/App.tsx]: Removed!

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**